Linker error when building rttr 0.9.6 with clang on OS X

To generate the ninja files I used: `cmake -G "Ninja"`
OS X version: 10.11.2
clang version:
```
$ clang --version
Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin15.2.0
Thread model: posix
``` 

Linker error:
```
[27/55] Linking CXX shared library bin/librttr_core.0.9.6.dylib
FAILED: : && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -frtti -std=c++11 -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG -frtti -g -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -dynamiclib -Wl,-headerpad_max_install_names  -o bin/librttr_core.0.9.6.dylib -install_name /Users/mgee/dev/rttr/build/bin/librttr_core.0.9.6.dylib src/rttr/CMakeFiles/rttr_core.dir/constructor.cpp.o src/rttr/CMakeFiles/rttr_core.dir/destructor.cpp.o src/rttr/CMakeFiles/rttr_core.dir/enumeration.cpp.o src/rttr/CMakeFiles/rttr_core.dir/method.cpp.o src/rttr/CMakeFiles/rttr_core.dir/parameter_info.cpp.o src/rttr/CMakeFiles/rttr_core.dir/policy.cpp.o src/rttr/CMakeFiles/rttr_core.dir/property.cpp.o src/rttr/CMakeFiles/rttr_core.dir/registration.cpp.o src/rttr/CMakeFiles/rttr_core.dir/type.cpp.o src/rttr/CMakeFiles/rttr_core.dir/variant.cpp.o src/rttr/CMakeFiles/rttr_core.dir/variant_array_view.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/misc/compare_equal.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/misc/compare_less.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/misc/standard_types.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/conversion/std_conversion_functions.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/constructor/constructor_wrapper_base.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/destructor/destructor_wrapper_base.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/enumeration/enumeration_helper.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/enumeration/enumeration_wrapper_base.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/method/method_wrapper_base.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/parameter_info/parameter_info_wrapper_base.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/property/property_wrapper_base.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/registration/registration_executer.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/type/type_database.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/type/type_register.cpp.o src/rttr/CMakeFiles/rttr_core.dir/detail/variant/variant_compare_less.cpp.o   && :
Undefined symbols for architecture x86_64:
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::data() const", referenced from:
      __GLOBAL__sub_I_standard_types.cpp in standard_types.cpp.o
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::size() const", referenced from:
      __GLOBAL__sub_I_standard_types.cpp in standard_types.cpp.o
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::c_str() const", referenced from:
      __GLOBAL__sub_I_standard_types.cpp in standard_types.cpp.o
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::empty() const", referenced from:
      __GLOBAL__sub_I_standard_types.cpp in standard_types.cpp.o
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::length() const", referenced from:
      __GLOBAL__sub_I_standard_types.cpp in standard_types.cpp.o
  "std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::operator[](unsigned long)", referenced from:
      __GLOBAL__sub_I_standard_types.cpp in standard_types.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

Temporary workaround to get it to link: Remove the reflection of the std::string functions by deleting [these lines](https://github.com/rttrorg/rttr/blob/425d528bf828703e6a6bf371172342d37a87c1f6/src/rttr/detail/misc/standard_types.cpp#L72-L87).

--- a/src/rttr/detail/misc/standard_types.cpp
+++ b/src/rttr/detail/misc/standard_types.cpp
@@ -32,6 +32,8 @@
 #include <list>
 #include <set>
 
+template class std::basic_string<char>;
+
 RTTR_REGISTRATION
 {
     using namespace rttr;

